app/web/src/features/dashboard/CommunityBrowser.tsx

```diff
@@ -0,0 +1,166 @@
+import {
+  Divider,
+  Link as MuiLink,
+  List,
+  ListItem,
+  ListItemText,
+} from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import CircularProgress from "components/CircularProgress";
+import {
+  useCommunity,
+  useListSubCommunities,
+} from "features/communities/hooks";
+import { LOAD_MORE } from "features/dashboard/constants";
+import { Community } from "proto/communities_pb";
+import { ReactNode, useState } from "react";
+import { Link } from "react-router-dom";
+import { routeToCommunity } from "routes";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    flexDirection: "row",
+    "& > * + *": {
+      marginInlineStart: theme.spacing(2),
+    },
+  },
+  list: {
+    minWidth: "10rem",
+  },
+  loader: {
+    margin: "auto",
+  },
+  selected: {
+    fontWeight: "bold",
+  },
+}));
+
+export default function CommunityBrowser() {
+  const classes = useStyles();
+  const [selected, setSelected] = useState<Community.AsObject[]>([]);
+
+  const globalCommunityQuery = useCommunity(1);
+
+  //react-query doesn't have useInfiniteQueries
+  //as a workaround, cache query results
+  //and only show "Load more" for last column
+  const query = useListSubCommunities(
+    selected?.[selected.length - 1]?.communityId || 1
+  );
+  const [cachedQueryResults, setCachedQueryResults] = useState<
+    {
+      data: Community.AsObject[];
+      hasMore?: boolean;
+    }[]
+  >([]);
+
+  const handleClick = (community: Community.AsObject, level: number) => {
+    if (level === selected.length) {
+      setCachedQueryResults([
+        ...cachedQueryResults,
+        {
+          data: query.data!.pages.flatMap((page) => page.communitiesList),
+          hasMore: query.hasNextPage,
+        },
+      ]);
+    } else {
+      setCachedQueryResults(cachedQueryResults.slice(0, level + 1));
+    }
+    setSelected([...selected.slice(0, level), community]);
+  };
+
+  return (
```

Comment:
How the heck does this know to hide the nested community columns when you switch to a different community at the higher level - e.g. going from Oceania (?) to Europe and back to Oceania knows to only show two columns again? Doesn't look immediately obvious to me where "the magic" from just staring at the code